CMakeLists.txt

```
@@ -149,9 +149,32 @@ if (BUILD_VERSION STREQUAL "0.0.0")
       file (STRINGS ${PROJECT_SOURCE_DIR}/VERSION_CURRENT BUILD_VERSION)
       message (STATUS "File VERSION_CURRENT contained BUILD_VERSION ${BUILD_VERSION}")
    else ()
-      set (RELEASE_BRANCH_REF "origin/r1.5")
-      include (GetVersion)
-      GetVersion (BUILD_VERSION)
+      find_package (PythonInterp)
+      if (PYTHONINTERP_FOUND)
+         execute_process (
+            COMMAND ${PYTHON_EXECUTABLE} etc/calc_release_version.py
+            WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+            OUTPUT_VARIABLE CALC_RELEASE_VERSION
+            RESULT_VARIABLE CALC_RELEASE_VERSION_RESULT
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+         )
+         if (NOT CALC_RELEASE_VERSION_RESULT STREQUAL 0)
+            # If python failed above, stderr would tell the user about it
+            message (FATAL_ERROR
+               "BUILD_VERSION not specified and could not be calculated\
+ (script invocation failed); specify in CMake command, -DBUILD_VERSION=<version>"
+            )
+         else ()
+            set (BUILD_VERSION ${CALC_RELEASE_VERSION})
+            message ("calculated BUILD_VERSION ${BUILD_VERSION}")
+            file (WRITE ${CMAKE_BINARY_DIR}/VERSION_CURRENT ${CALC_RELEASE_VERSION})
```

```suggestion
            file (WRITE ${PROJECT_SOURCE_DIR}/VERSION_CURRENT ${CALC_RELEASE_VERSION})
```

To match the check on L148.

---

@kevinAlbs So, I think I left a small gap here.  The `PROJECT_SOURCE_DIR` location was original to the C driver.  When implemented `calc_release_version.py` in the C++ driver, we allowed for `PROJECT_SOURCE_DIR`, but since it did not have a history of `VERSION_CURRENT` like the C driver, we preferred the `CMAKE_BINARY_DIR` location instead.  That way, clearing directory where the build system was generated would also eliminate the `VERSION_CURRENT` file.  I think that for libmonogocrypt we are best served preferring `CMAKE_BINARY_DIR`, so I'm proposing instead to do this:

``` set (BUILD_VERSION "0.0.0" CACHE STRING "Library version")
 if (BUILD_VERSION STREQUAL "0.0.0")
-   if (EXISTS ${PROJECT_SOURCE_DIR}/VERSION_CURRENT)
+   if (EXISTS ${CMAKE_BINARY_DIR}/VERSION_CURRENT)
+      file (STRINGS ${CMAKE_BINARY_DIR}/VERSION_CURRENT BUILD_VERSION)
+   elseif (EXISTS ${PROJECT_SOURCE_DIR}/VERSION_CURRENT)
       file (STRINGS ${PROJECT_SOURCE_DIR}/VERSION_CURRENT BUILD_VERSION)
       message (STATUS "File VERSION_CURRENT contained BUILD_VERSION ${BUILD_VERSION}")
    else ()
```

This would match the "better" approach we implemented in the C++ driver.

---

Using `CMAKE_BINARY_DIR` SGTM.